### PR TITLE
KOGITO-3309: changed parent of kogito-bom to not inherit build requir…

### PR DESCRIPTION
**Pull request by mareknovotny**
…ed dependency management

https://issues.redhat.com/browse/KOGITO-3309

kogito-bom inherits kogito-build-parent dependencyManagement so having kogito-bom only as user BOM i switched the parent of it.

Followup PR in kogito-examples https://github.com/kiegroup/kogito-examples/pull/405

### CODE REVIEW

File: kogito-build-parent/pom.xml [hunk 1]
```diff
@@ -5,13 +5,15 @@
   <modelVersion>4.0.0</modelVersion>
 
   <parent>
-    <groupId>org.kie.kogito</groupId>
-    <artifactId>kogito-runtimes</artifactId>
-    <version>1.0.0-SNAPSHOT</version>
-    <relativePath>../pom.xml</relativePath>
+    <groupId>org.jboss</groupId>
```
Comment (rsynek) [inline kogito-build-parent/pom.xml:8]: I don't understand the change - kogito-bom's parent is still this pom.xml (kogito-build-parent), which defines versions of many 3rd party dependencies.

I would rather expect:
- keeping the kogito-runtimes as a parent of the kogito-build-parent
- making kogito-runtimes a parent of the kogito-bom
- importing the kogito-bom into the kogito-build-parent

**Review by ge0ffrey — CHANGES_REQUESTED** [review]
I agree with Radovan:
- kogito-parent extends jboss-parent and has no dependencies.
- kogito-bom extends kogito-parent and only has kie dependencies in depMgmt, no external ones.
- kogito-build-parent extends kogito-parent, imports kogito-bom and has all external dependencies in depMgmt
- every other module extends kogito-build-parent.

**Comment by danielezonca** [comment]
Note, this PR is going to add a new addon so it should be updated accordingly with this new refactor
https://github.com/kiegroup/kogito-runtimes/pull/807

**Comment by mareknovotny** [comment]
Jenkins retest this please

File: integration-tests/integration-tests-springboot/src/it/integration-tests-springboot-kafka-it/pom.xml [hunk 2]
```diff
@@ -99,6 +99,7 @@
         <dependency>
             <groupId>org.kie.kogito</groupId>
             <artifactId>kogito-test-utils</artifactId>
+            <version>${project.version}</version>
```
Comment (mareknovotny) [inline integration-tests/integration-tests-springboot/src/it/integration-tests-springboot-kafka-it/pom.xml:102]: kogito-bom doesn'ŧ contain kogito-test-utils declaration, should it or is that just for internal usage? @danielezonca @cristianonicolai  @ruromero

Comment (danielezonca) [reply integration-tests/integration-tests-springboot/src/it/integration-tests-springboot-kafka-it/pom.xml:102]: As far as I know that module is intended to be used only for internal usage so I'm fine to keep it outside kogito-bom

Comment (mareknovotny) [reply integration-tests/integration-tests-springboot/src/it/integration-tests-springboot-kafka-it/pom.xml:102]: thanks, then this is ok and kogito-examples need to have depMgmt too

**Comment by mareknovotny** [comment]
Jenkins retest this

**Review by rsynek — APPROVED** [review]
LGTM, thank you!

**Comment by mareknovotny** [comment]
> Note, this PR is going to add a new addon so it should be updated accordingly with this new refactor
> #807

then better to merge this first, all green now

**Comment by mareknovotny** [comment]
merge it then @kiegroup/gatekeepers (hold my hands off from it ;) )

**Review by ge0ffrey — APPROVED** [review]
Great work!

**Comment by ge0ffrey** [comment]
@evacchi Can we have this merged?

**Pull request merged**